src/mds/MDSAuthCaps.cc

```diff
@@ -373,6 +373,89 @@ bool MDSAuthCaps::parse(CephContext *c, std::string_view str, ostream *err)
   }
 }
 
+bool MDSAuthCaps::merge(MDSAuthCaps newcap)
+{
+  assert(newcap.grants.size() == 1);
```

Comment:
I don't know the `mds` code base and its conventions. In other parts typically `ceph_assert()` is used. Is `assert()` intentional here?

Reply:
I talked with Patrick about. It appears that using `ceph_assert()` is better.